### DISCUSSION TITLE
Which .NET Framework for Plugins?

### DISCUSSION THREAD

**Discussion by verybadsoldier** [Rem0o/FanControl.Releases#1874] [General]
Hi everyone,

when developing a plugin for this, which target framework to use? I got it running using ".NET Framework", but didn't work with ".NET". Is it correct that only ".NET Framework" can be used for plugins?

**Comment by EvanMulawski** [comment]
Target **.NET Standard 2.0** (`netstandard2.0`) if possible. If that isn't possible (e.g. due to dependencies or .NET Framework-specific APIs), release two builds: one for **.NET Framework 4.8** (`net48`) and one for **.NET 7** (`net7.0`).

**Reply by verybadsoldier** [reply]
Ok, thank you. But why two releases? It seems that FanControl is using .NET Framework 4.8, so it would not be able to load a .NET 7 assembly anyway? So what would be the use case for a .NET 7 plugin?
That is at least the impression I got after googling a bit.

**Reply by EvanMulawski** [reply]
There are two builds of Fan Control: one targets .NET Framework 4.8 and the other targets .NET 7.

![image](https://github.com/Rem0o/FanControl.Releases/assets/1044791/f3c6dd44-54d7-4967-ab57-89bd761b2ffa)

**Reply by verybadsoldier** [reply]
Oh hell, I wasn't aware of that. Many thanks! That explains .... just everything
